src/PluginClient.php

```diff
@@ -62,9 +65,20 @@ public function __construct($client, array $plugins = [], array $options = [])
      */
     public function sendRequest(RequestInterface $request)
     {
-        $promise = $this->sendAsyncRequest($request);
+        if (!($this->client instanceof HttpClient)) {
+            return $this->sendAsyncRequest($request)->wait();
+        }
+
+        $client = $this->client;
+        $pluginChain = $this->createPluginChain($this->plugins, function (RequestInterface $request) use ($client) {
```

Comment:
Yeah, but I still can't see the difference. Here is what happens now:

If the client is async (not nonasync) then the async version is called.

If it is sync, we emulate an async one. In the plugin clien's sync call we wrap the sync client call with the same code as we did in case of the emulated client. Same code, whats the difference apart from the method name?